components/storage/indexeddb/mod.rs

```diff
@@ -194,10 +197,64 @@ impl<E: KvsEngine> IndexedDBEnvironment<E> {
     }
 }
 
+fn backend_error_from_sqlite_error(err: RusqliteError) -> BackendError {
+    if is_sqlite_disk_full_error(&err) {
+        BackendError::QuotaExceeded
+    } else {
+        BackendError::DbErr(format!("{err:?}"))
+    }
+}
+
+fn is_sqlite_disk_full_error(error: &RusqliteError) -> bool {
```

Comment:
Seems to me like this could be useful for other parts of storage, so maybe put it in `https://github.com/servo/servo/blob/main/components/storage/shared/mod.rs or elsewhere in `storage`? cc @janvarga

Reply:
moved it